src/generic_factory.h

```diff
@@ -117,6 +117,12 @@ class string_id_reader;
 template<typename T>
 class generic_factory
 {
+
+	public:
+        virtual ~generic_factory()
+        {
```

Comment:
As the function body is empty, it should be defaulted: `virtual ~foo() = default;`.